### PR TITLE
Allure steps for bucket_list tests.

### PR DESCRIPTION
**Pull request by MaxGelbakhiani**
Part of #39

A set of bucket_list S3 tests wrapped with allure steps.

### CODE REVIEW

**Review by vvarg229 — APPROVED** [review]

File: s3tests_boto3/functional/test_s3.py [hunk 1]
```diff
@@ -1843,20 +2075,25 @@ def test_bucket_delete_nonempty():
...
 
     e = assert_raises(ClientError, client.delete_bucket, Bucket=bucket_name)
-
+    allure.attach(f"Raise assertion:\n{e}")
     status, error_code = _get_status_and_error_code(e.response)
     assert status == 409
     assert error_code == "BucketNotEmpty"
 
 
+@allure.step("Do set bucket canned ACL")
```
Comment (roman-khimov) [inline s3tests_boto3/functional/test_s3.py:2084]: `s/Do //`?

File: s3tests_boto3/functional/test_s3.py [hunk 2]
```diff
@@ -1869,6 +2106,7 @@ def _do_set_bucket_canned_acl_concurrent(client, bucket_name, canned_acl, num, r
     return t
 
 
+@allure.step("Do wait completion")
```
Comment (roman-khimov) [inline s3tests_boto3/functional/test_s3.py:2109]: Wait for completion?

File: s3tests_boto3/functional/test_s3.py [hunk 3]
```diff
@@ -1843,20 +2075,25 @@ def test_bucket_delete_nonempty():
...
+        req_params=dict(ACL=canned_acl, Bucket=bucket_name)
+        with allure.step(f"Put bucket ACL {req_params}"):
+            client.put_bucket_acl(**req_params)
         results[i] = True
     except:
         results[i] = False
+    allure.attach(f"Results:\n{results}")
 
 
+@allure.step("Do set ACL concurrent")
```
Comment (roman-khimov) [inline s3tests_boto3/functional/test_s3.py:2096]: `s/Do //`?

File: s3tests_boto3/functional/test_s3.py [hunk 4]
```diff
@@ -905,7 +986,10 @@ def test_bucket_listv2_prefix_basic():
     bucket_name = _create_objects(keys=key_names)
     client = get_client()
 
-    response = client.list_objects_v2(Bucket=bucket_name, Prefix="foo/")
+    req_params = dict(Bucket=bucket_name, Prefix="foo/")
+    with allure.step(f"Request list_objects: {req_params}"):
```
Comment (evgeniiz321) [inline s3tests_boto3/functional/test_s3.py:990]: it is better to wrap client.list_objects_v2 and similar methods with an allure specific code, cause too many duplicates here

Comment (evgeniiz321) [reply s3tests_boto3/functional/test_s3.py:990]: perhaps it is even worth to just create a wrapper for a client class with getattribute overridden, and intercept there every call to methods and log input params and output from them to allure

Comment (MaxGelbakhiani) [reply s3tests_boto3/functional/test_s3.py:990]: Opened a new PR #57 with client class methods overridden. 
Closing this one.

**Pull request closed without merging**